src/internet_identity_interface/src/internet_identity/device_data.rs

```diff
@@ -0,0 +1,22 @@
+use crate::internet_identity::types::{DeviceData, DeviceProtection, KeyType, Purpose};
+use candid::Principal;
+use serde_bytes::ByteBuf;
+
+impl DeviceData {
+    pub fn principal(&self) -> Principal {
+        Principal::self_authenticating(&self.pubkey)
+    }
+
+    /// Creates a new device that can be used for test purposes.
+    pub fn test_device() -> Self {
+        Self {
+            pubkey: ByteBuf::from([0; 32]),
+            alias: "test device".to_string(),
+            credential_id: None,
+            purpose: Purpose::Authentication,
```

Comment:
nit: maybe add something about `auth` to the helper name, say `test_device_auth` or `auth_test_device`?
(The default of `Purpose::Authentication` is somewhat arbitrary, and it is in fact the second value of the `Purpose`-enum)

Reply:
Makes sense 👍